src/config.rs

```diff
@@ -12,6 +12,8 @@ pub struct Config {
     #[serde(default)]
     pub cargo_target_dir: Option<String>,
     #[serde(default)]
+    pub preserved_project_dir: Option<String>,
```

Comment:
```suggestion
    pub cargo_target_dir: Option<PathBuf>,
    #[serde(default)]
    pub preserved_project_dir: Option<PathBuf>,
```
You can swap these two settings, So you don't need to convert them by yourself since Serde gonna take care of it for us.